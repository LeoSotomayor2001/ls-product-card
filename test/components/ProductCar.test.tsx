
import  renderer  from 'react-test-renderer';
import { ProductCar } from '../../src/components';
import React from 'react';
import { product1 } from '../data/products';


const {act}=renderer;

describe('ProductCar', () => {
  test('debe de mostrar el componente correctamente ', () => {
    
    const wraper=renderer.create(
          <ProductCar product={product1}>
              {
                ()=> (
                  <h1>Hola ProductCar</h1>
                )
              }
          </ProductCar>
    );

    expect(wraper.toJSON()).toMatchSnapshot();
      
  });
  
  test('debe de incrementar el contador', () => {
      const wraper=renderer.create(
          <ProductCar product={product1}>
              {
                ({count,increaseBy})=> (
                  <>
                    <h1>Product CARD</h1>

                    <span>{count}</span>
                    <button onClick={()=> increaseBy(1)}></button>
                  
                  </>
                )
              }
          </ProductCar>
      );

      let tree=wraper.toJSON();
      expect(tree).toMatchSnapshot();

      act(()=>{

        (tree as any).children[2].props.onClick();
      })
      
      
      tree=wraper.toJSON();

      expect( (tree as any).children[1].children[0] ).toBe('1');
      
  })
  

  
})

