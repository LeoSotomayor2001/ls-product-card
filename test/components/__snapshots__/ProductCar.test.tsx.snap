// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCar debe de incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product CARD
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCar debe de mostrar el componente correctamente  1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Hola ProductCar
  </h1>
</div>
`;
